Geometry_Engine/Query/Bounds.cs

```diff
@@ -400,30 +402,106 @@ public static BoundingBox Bounds(this CompositeGeometry group)
             if (elements.Count == 0)
                 return null;
 
-            BoundingBox box = elements[0].IBounds();
+            BoundingBox box = Bounds(elements[0] as dynamic);
             for (int i = 1; i < elements.Count; i++)
-                box += elements[i].IBounds();
+                box += Bounds(elements[i] as dynamic);
 
             return box;
         }
 
+        /******************************************/
+        /****            IElement0D            ****/
+        /******************************************/
 
-        /***************************************************/
-        /**** Public Methods - Interfaces               ****/
-        /***************************************************/
+        public static BoundingBox Bounds(this IElement0D element0D)
+        {
+            return Bounds(element0D.IGeometry());
+        }
+
+
+        /******************************************/
+        /****            IElement1D            ****/
+        /******************************************/
 
-        public static BoundingBox IBounds(this IGeometry geometry)
+        public static BoundingBox Bounds(this IElement1D element1D)
         {
-            if (geometry == null)
+            return IBounds(element1D.IGeometry());
+        }
+
+
+        /******************************************/
+        /****            IElement2D            ****/
+        /******************************************/
+
+        public static BoundingBox Bounds(this IElement2D element2D)
+        {
+            List<ICurve> elementCurves = element2D.ElementCurves(true);
+
+            if (elementCurves.Count == 0)
                 return null;
 
-            return Bounds(geometry as dynamic);
+            BoundingBox box = IBounds(elementCurves[0]);
+            for (int i = 1; i < elementCurves.Count; i++)
+                box += IBounds(elementCurves[i]);
+
+            return box;
+        }
+
+
+        /******************************************/
+        /****        Interface methods         ****/
+        /******************************************/
+
+        public static BoundingBox IBounds(this IObject element)
+        {
+            if (element == null)
+                return null;
+
+            if (element is IGeometry || element is IElement)
+                return Bounds(element as dynamic);
+            return null;
+        }
+
+        /******************************************/
+
+        public static BoundingBox Bounds(this IGeometry element)
+        {
+            if (element == null)
+                return null;
+
+            return Bounds(element as dynamic);
+        }
+
+        /******************************************/
+
+        public static BoundingBox Bounds(this IElement element)
+        {
+            if (element == null)
+                return null;
+
+            return Bounds(element as dynamic);
+        }
```

Comment:
The lower two methods used to be `IBounds` but things like `ICurve` are both `IGeometry` and `IElement` so that broke, which is why I added the top method

Reply:
Your proposal does not look good, but might be the only way to go - I would only recommend making `Bounds(IGeometry)` and `Bounds(IElement)` `private` as they are not compliant (their name should be `IBounds`).

We can flag it up and discuss in a wider group once the PR is ready.

Reply:
I agree it does look quite horrendous. 

But a note is that I was thinking of holding of on this PR until #1319 and/or #1314 is merged since they all mess with the `.csproj`.

But if we want this merged soon I could probably make it ready today, but I'm expecting that there might be a lot of dependent repos that'll need updating, which would slow things down.

Reply:
Yeah I would definitely not rush with that one as long as there is no legitimate urgency. @rwemay @al-fisher

Reply:
Can you explain the point of having `IBounds(this IObject element)` ? I cannot think of a case where objects would be handled not as a list of `IGeometry` or a list of `IElement` but as a list of both.  
This also feels very wrong to explicitely provide a method for all `IObject` while you are actually only covering 2 specific types internally.

Reply:
`ICurve`s are both `IElement` and `IGeometry`, hence a list of `ICurve`s are a list of both. 

This is some initial work before I realized the scope of the problem and is really not the solution I would go for.